msibi/optimize.py

```diff
@@ -69,58 +72,61 @@ def __init__(self, rdf_cutoff, n_rdf_points, pot_cutoff=None, r_switch=None,
             pot_cutoff = rdf_cutoff
         self.pot_cutoff = pot_cutoff
         # TODO: Describe why potential needs to be messed with to match the RDF.
-        self.pot_r = np.arange(0.0, self.pot_cutoff + self.dr, self.dr)
+        if self.engine == 'hoomd': 
+            self.pot_r = np.arange(0.0, self.pot_cutoff + self.dr, self.dr)
+        elif self.engine == 'lammps': # LAMMPS starts potential files at dr, not 0
+            self.pot_r = np.arange(self.dr, self.pot_cutoff + self.dr, self.dr) 
+        else:
+            raise UnsupportedEngine(engine)
 
         if not r_switch:
             r_switch = self.pot_r[-5]
         self.r_switch = r_switch
 
-    def optimize(self, states, pairs, n_iterations=10, engine='hoomd',
-                 start_iteration=0):
+    def optimize(self, states, pairs, n_iterations=10, start_iteration=0):
         """
         """
         for pair in pairs:
             for state, data in pair.states.items():
                 if len(data['target_rdf']) != self.n_rdf_points:
                     raise ValueError('Target RDF in {} of pair {} is not the '
-                                     'same length as n_rdf_points.'.format(
-                        state.name, pair.name))
+                                     'same length as n_rdf_points; {} vs {}.'.format(
+                        state.name, pair.name, len(data['target_rdf']), self.n_rdf_points))
         self.states = states
         self.pairs = pairs
         self.n_iterations = n_iterations
-        self.initialize(engine=engine)
+        self.initialize()#(engine=self.engine)
 
         for n in range(start_iteration + self.n_iterations):
             logging.info("-------- Iteration {n} --------".format(**locals()))
-            run_query_simulations(self.states, engine=engine)
-            self._update_potentials(n, engine)
+            run_query_simulations(self.states, engine=self.engine)
+            self._update_potentials(n)
 
-    def _update_potentials(self, iteration, engine):
+    def _update_potentials(self, iteration):
         """Update the potentials for each pair. """
         for pair in self.pairs:
             self._recompute_rdfs(pair, iteration)
             pair.update_potential(self.pot_r, self.r_switch)
             pair.save_table_potential(self.pot_r, self.dr, iteration=iteration,
-                                      engine=engine)
+                                      engine=self.engine)
 
     def _recompute_rdfs(self, pair, iteration):
         """Recompute the current RDFs for every state used for a given pair. """
         for state in pair.states:
-            pair.compute_current_rdf(state, self.rdf_r_range,
-                                     n_bins=self.rdf_n_bins,
+            pair.compute_current_rdf(state, self.rdf_r_range/10,
+                                     n_bins=self.rdf_n_bins-1,
```

Comment:
I was sloppy with nm vs A. I wrote it assuming lammps units used A whereas mdtraj uses nm. To me, it would be counterintuitive to initialize a lammps-centric system while passing nm distances. Not sure the best way to handle this generally. 

I think I used rdf_n_bins-1 because the lammps potential excludes r = 0 and the length of the potential and rdf need to match. The system I was using has a target rdf with 100 points and was writing potential files with 99 points.